python/paddle/fluid/core.py

```diff
@@ -57,27 +56,112 @@
 except Exception as e:
     raise e
 
-load_noavx = False
 
-has_avx = False
-if sys.platform == 'darwin':
-    try:
-        has_avx = os.popen('sysctl machdep.cpu.features | grep -i avx').read(
-        ) != ''
-    except Exception as e:
-        sys.stderr.write(
-            'Can not get the AVX flag from machdep.cpu.features.\n')
-    if not has_avx:
+def avx_supported():
+    """
+    Whether current system(Linux, MacOS, Windows) is supported with AVX.
+    """
+    import platform
+    from .. import compat as cpt
+    sysstr = platform.system().lower()
+    has_avx = False
+    if sysstr == 'linux':
+        try:
+            has_avx = os.popen('cat /proc/cpuinfo | grep -i avx').read() != ''
+        except Exception as e:
+            sys.stderr.write('Can not get the AVX flag from /proc/cpuinfo.\n'
+                             'The original error is: %s\n' %
+                             cpt.get_exception_message(e))
+        return has_avx
+    elif sysstr == 'darwin':
         try:
             has_avx = os.popen(
-                'sysctl machdep.cpu.leaf7_features | grep -i avx').read() != ''
+                'sysctl machdep.cpu.features | grep -i avx').read() != ''
         except Exception as e:
             sys.stderr.write(
-                'Can not get the AVX flag from machdep.cpu.leaf7_features.\n')
-else:
-    has_avx = 'avx' in get_cpu_info()['flags']
+                'Can not get the AVX flag from machdep.cpu.features.\n'
+                'The original error is: %s\n' % cpt.get_exception_message(e))
+        if not has_avx:
+            try:
+                has_avx = os.popen(
+                    'sysctl machdep.cpu.leaf7_features | grep -i avx').read(
+                    ) != ''
+            except Exception as e:
+                sys.stderr.write(
+                    'Can not get the AVX flag from machdep.cpu.leaf7_features.\n'
+                    'The original error is: %s\n' %
+                    cpt.get_exception_message(e))
+        return has_avx
+    elif sysstr == 'windows':
+        import ctypes
```

Comment:
will this package be an dependency?

Reply:
It seems in `The Python Standard Library` https://docs.python.org/3.6/library/ctypes.html